src/test/java/study/step4/CarTest.java

```diff
@@ -0,0 +1,48 @@
+package study.step4;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class CarTest {
+
+    @ParameterizedTest
+    @DisplayName("차량 이름 미 입력시 예외처리 테스트")
+    @NullAndEmptySource
+    void noNameCarMakeExceptionTest(String carName){
+        assertThatThrownBy(() ->{
+            new Car(carName);
+        }).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @ParameterizedTest
+    @DisplayName("차량 포지션 기준에 부합하지 않을시 (0보다 작을시) 예외처리 테스트")
+    @ValueSource(ints = {-1, -2})
+    void noPositionCarMakeExceptionTest(int carPosition){
+        assertThatThrownBy(() ->{
+            new Car("", carPosition);
+        }).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @ParameterizedTest
+    @DisplayName("조건 충족시 차량 포지션 증가 체크 : ")
+    @ValueSource(ints = {4,5,6,7,8,10})
+    void moveMethodTest(int number){
+        CarMoveCondition mockRandomGenerator = new MockRandomCarMoveCondition(number);
+        Car car = new Car("랜덤카").move(mockRandomGenerator);
```

Comment:
```suggestion
        Car car = new Car("랜덤카").move(() -> true);
```
Mock객체를 이용하여 일반적으로 테스트 하기 힘든 랜덤값과 같은 메소드 결과를 테스트 해주신 부분은 정말 좋았습니다.
한가지 팁도 드리자면 람다를 활용하여 테스트 할 수도 있어요 !